### PR TITLE
add doGet, create Comment class, styling, updating comment interface in index.html...

### PR DESCRIPTION
**Pull request by lexidelorey**
update list element function to work with Comment class

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -33,28 +39,47 @@
 public class DataServlet extends HttpServlet {
 
   private static final String COMMENT_ENTITY_KEY = "Comment";
-  private static final String COMMENT_PROPERTY_NAME = "comment";
-  private static final String TIME_PROPERTY_NAME = "dateTime";
+  private static final String COMMENT_PROPERTY_NAME = "userComment";
+  private static final String TIME_PROPERTY_NAME = "dateTimeCreated";
```
Comment (rglassett) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:43]: I agree that `"dateTimeCreated"` is a better name than `"dateTime"`--but why change `"comment"`?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -33,28 +39,47 @@
...
   
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(COMMENT_ENTITY_KEY)
+                  .addSort("dateTimeCreated", SortDirection.DESCENDING);
```
Comment (rglassett) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Replace `"dateTimeCreated"` with `TIME_PROPERTY_NAME`. If you reference a nonexistent field, you get a compiler error. If you misspell a String, you won't know anything is wrong until runtime at best--and in more complicated code, the error can be hidden by other parts of your application logic.

This is an anti-pattern known as a "magic string". See [this StackOverflow post](https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings) for a more detailed breakdown.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -33,28 +39,47 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String text = request.getParameter(COMMENT_PROPERTY_NAME);
-    long time = System.currentTimeMillis();
+    String userComment = request.getParameter("comment");
```
Comment (rglassett) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: 👍 : I agree that your request param names are and should be distinct from your data store's property names.
👎 : It's still better to reference a constant than a magic string.

Maybe `COMMENT_PARAM_NAME`?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -33,28 +39,47 @@
...
     SimpleDateFormat format= new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
-    Date dateTime = new Date(System.currentTimeMillis());
+    Date dateTimeCreated = new Date(System.currentTimeMillis());
 
     Entity commentEntity = new Entity(COMMENT_ENTITY_KEY);
-    commentEntity.setProperty(COMMENT_PROPERTY_NAME, text);
-    commentEntity.setProperty(TIME_PROPERTY_NAME, dateTime);
-
+    commentEntity.setProperty("userComment", userComment);
+    commentEntity.setProperty("dateTimeCreated", dateTimeCreated);
```
Comment (rglassett) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:77]: Use constants for property names.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -6,7 +6,7 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
   </head>
-  <body onload = "rotateImages()">
+  <body onload = "rotateImages()";"getCommentsUsingArrows()">
```
Comment (rglassett) [inline portfolio/src/main/webapp/index.html:9]: I don't believe this will work as you expect it to. HTML attributes are specified by:

```
<tagname attributename="attributevalue">
```

In this case, the value of `onload` is `rotateImages()`. The value begins after the equals sign and opening double-quote, and ends when a matching double-quote is reached. If you want to execute more than one statement in JavaScript, you can use what's called an ["immediately invoked function expression" (IIFE)](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).

tl;dr:

```
<body onload="(function () {
  rotateImages();
  getCommentsUsingArrows();
})();">

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -80,10 +80,9 @@ <h2>Comments</h2>
       <form action="/data" method="post" target="_self">
         <textarea name="comment" placeholder="Comment..."></textarea>
         <br/><br/>
-        <input type="submit" value="Submit">
+        <input type="submit" value="Submit" >
       </form>
-      <p>Click below to see comments left by others!</p>
-      <button class="button" onclick="getCommentsUsingArrows()">Comments</button>
+        <script>getCommentsUsingArrows()</script>
```
Comment (rglassett) [inline portfolio/src/main/webapp/index.html:85]: If you make the "IIFE" change I recommended above, I don't think you'll need this line.

If it does provide value, then nit: indentation should match the lines above and below.

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -80,10 +80,9 @@ <h2>Comments</h2>
       <form action="/data" method="post" target="_self">
         <textarea name="comment" placeholder="Comment..."></textarea>
         <br/><br/>
-        <input type="submit" value="Submit">
+        <input type="submit" value="Submit" >
```
Comment (rglassett) [inline portfolio/src/main/webapp/index.html:83]: Nit: It's a good practice to avoid making non-functional (e.g. formatting) changes. Many projects apply an auto-formatter to their code, not only because the result is more consistent, but also because code review is easier if all diffs are semantically meaningful.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -55,18 +55,23 @@ function rotateImages() {
 }
 
 function getCommentsUsingArrows() {
```
Comment (rglassett) [inline portfolio/src/main/webapp/script.js:57]: Is this function name accurate? It's unclear to me what "usingArrows" means--it looks like you're simply waiting for the page to load, then fetching all the comments and inserting them into the document body.

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -55,18 +55,23 @@ function rotateImages() {
...
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const userCommentElement = document.createElement('span')
+  userCommentElement.innerText = comment.comment;
```
Comment (rglassett) [inline portfolio/src/main/webapp/script.js:73]: Variable naming could be improved. What is a `"userComment"` and how is it different from a `"comment"`?

In theory, all comments are user-submitted. This looks to me more like a comment "body" or "message".

File: portfolio/src/main/webapp/style.css [hunk 10]
```diff
@@ -15,6 +15,12 @@ h2 {
   font-family: Helvetica;
 }
 
+h3 {
+  font-size: 25px;
+  color: rgba(128, 128, 128, 0.24);
+  font-family: Helvetica;
+}
```
Comment (rglassett) [inline portfolio/src/main/webapp/style.css:22]: You probably don't have to worry about this for now, given the small and self-contained nature of the project--but in practice you should almost always prefer [class selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors) to [type selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Type_selectors).

The reasons are somewhat involved, but if you're interested and/or expect to write a lot of CSS, I would be happy to elaborate further.

File: portfolio/src/main/java/com/google/sps/data/Comments.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package com.google.sps.data;
+import java.util.Date;
+
+public final class Comments {
```
Comment (rglassett) [inline portfolio/src/main/java/com/google/sps/data/Comments.java:4]: This class looks to represent a single comment entity, not a plurality--please rename to `Comment`).

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (lexidelorey) [reply portfolio/src/main/webapp/script.js:57]: In my original version I had tested two versions -- arrows and async/await but I ended up going with arrows (=>) but you're right, it needs a more clear name now that there aren't two functions doing the same thing!

File: portfolio/src/main/webapp/style.css (hunk 10 above)
Comment (lexidelorey) [reply portfolio/src/main/webapp/style.css:22]: I'm not sure how much CSS I'll be writing, but I am always interested in learning new stuff, if you have time to explain!

File: portfolio/src/main/java/com/google/sps/data/Comments.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+  private long id;
+  private String comment;
+  private Date dateTime;
+
+  public Comments(long id, String comment, Date dateTime) {
+    this.id = id;
+    this.comment = comment;
+    this.dateTime = dateTime;
+  }
+}
```
Comment (usaeed-w) [inline portfolio/src/main/java/com/google/sps/data/Comments.java:15]: Nit: Add a new line for end of file.

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (rglassett) [reply portfolio/src/main/webapp/script.js]: I see! I would say that good function names describe what they do, not necessarily how they do it. Here I might go with `fetchAndDisplayComments()`. The fact that it's written as an arrow function is an implementation detail that does not (to me) seem interesting to the function's callers.

File: portfolio/src/main/webapp/style.css (hunk 10 above)
Comment (rglassett) [reply portfolio/src/main/webapp/style.css:22]: The basic gist is that CSS rules are effectively global. Once an HTML document loads a stylesheet, every rule declared in that stylesheet is applied to the entire document. There is no way to narrow the scope of the stylesheet to some sub-tree of the DOM.

The problem with type selectors is that they are overly general. An `h3` rule will apply to every `<h3>` on the page. But generally, an element's style depends on the surrounding context. An `<h3>` in the main content of an article will look different from an `<h3>` in a side nav, pop-up dialog, or block quote. Class selectors offer a way to achieve this by selectively targeting specific elements.

In a small single-page app, primarily using type selectors will probably serve you just fine (if it ain't broke, don't fix it). Just know that it's a best practice to use class selectors, especially in larger projects.

Comment (lexidelorey) [reply portfolio/src/main/webapp/style.css:22]: Ah, ok. That makes sense! Thank you for explaining that to me, I appreciate it!
